lua/folding.lua

```diff
@@ -35,7 +35,7 @@ function M.setup_plugin()
 
 
       if server_supports_folding then
-        client.config.callbacks['textDocument/foldingRange'] = M.fold_callback
+        client.config.handlers['textDocument/foldingRange'] = M.fold_handler
```

Comment:
To ensure backward-compat, we can keep adding the fold handler to the `callbacks` dict if it exists.

Reply:
I assume `handlers` did not exist when `callbacks` existed and thus needs a guard too right?

Reply:
Not 100%, but I believe that's the case. We could also just use an if/else if we want them to be exclusive. I'm fine with either way

Reply:
I don't thinks `callbacks` and `handlers` are mutually exclusive as `neovim` also goes through some form of deprecation process (see https://github.com/neovim/neovim/blob/d8c69adbabe963142f433a2ddad172ff46413f15/runtime/lua/vim/lsp.lua#L553-L554). Current approach looks good to me.